### ISSUE TITLE
pyjade not compatible with itertools.groupby (head item is lost)

### ISSUE THREAD

**Issue by kavinyao** [syrusakbary/pyjade#110]
I am using pyjade to generate jinja2 templates and it's awesome! However, I found a problem with `__pyjade_iter` when used together with some iterators (`itertools.groupby` in my case). The result is that the first item is _never_ rendered in the output.

After some investigation, I found that in the `iteration` function would retrieve the first item of `obj` and return `obj` directly if the cardinality equals the key number. However, the problem arises when the `obj` _itself_ is an iterator (the `__iter__` function returns `self`). In this case, the first item (`head`) will be lost. Many useful classes in [itertools](http://docs.python.org/2/library/itertools.html) are implemented like this and are meant to be iterated only once. So, the `iteration` function, by applying `head = next(iter(obj), None)`, breaks the semantic of such classes.

Basically, I suppose no modification should be made to `obj` for the sake of safety. However, I understand that to implement jade's for loop with index, you have to do something like this.

One simple fix is to change `return obj` on [this line](https://github.com/SyrusAkbary/pyjade/blob/master/pyjade/runtime.py#L97) to 

``` python
if id(obj) == id(iter(obj)):
    # preserve the head item of one-off iterators
    return itertools.chain([head], obj)
else:
    return obj
```

Hope there's better fix.

(Sorry for the verbosity. :stuck_out_tongue_winking_eye:)
